core/lib/spree/preferences/static_model_preferences.rb

```diff
@@ -27,20 +22,30 @@ def []=(key, value)
         def to_hash
           @preferences.deep_dup
         end
+
+        delegate :keys, to: :@preferences
       end
 
       def initialize
-        @store = Hash.new do |data, klass|
-          data[klass] = {}
-        end
+        @store = {}
```

Comment:
I do think it's cleaner to create the Hash with a fresh hash as the default value, for a few reasons:

- Otherwise, we need to sanitize it every time we access a key, as we're doing now twice in `#add` & `#fetch`.
- Could break something in edge cases where code relies on returning a Hash.
- In my view, it communicates better as it nails down the data structure we need.

Thoughts?

Reply:
The reason I touched this is that while storing prefs for solidus_stripe I noticed that trying a non existing key would add the key to the hash, e.g.:

```
>> sp = Spree::Preferences::StaticModelPreferences.new
=> #<Spree::Preferences::StaticModelPreferences:0x00000001124dac70 @store={}>
>> sp.for_class("Foo")
=> {}
>> sp
=> 
#<Spree::Preferences::StaticModelPreferences:0x00000001124dac70
 @store={"Foo"=>{}}>          
>> 
```

I'll change it to a simpler `Hash.new { {} }` so it behaves this way:

```
>> h = Hash.new { {} }
=> {}
>> h[:asdf]
=> {}
>> h
=> {}
>> h[:asdf] = 1234
=> 1234
>> h
=> {:asdf=>1234}
>> 
```

Reply:
I had to walk back on this, my second example wasn't correct, and after all we only need to set the empty hash in `#add` and fallback to a default in `#for_class` which is not much.

The whole preferences system could probably be simplified but didn't want to launch into a campaign, and this will at least provide a consistent the experience within initializers.

Reply:
Hmm, I see. I never realized that Ruby had this behavior. What about the change in the API (i.e., returning `nil` instead of `Hash`? Do you think it's plausible it could break stores? cc @kennyadsl

Reply:
I'd say that the whole API a bit sparse and could be streamlined, it also only serves one model inside solidus (although it's probably more across the ecosystem). But I wanted to surgically solve only the problem of not being able to pass strings for class names and finally make it consistent with other parts of the configuration.

There's also probably a case for making some of these APIs to be private and so not susceptible to backward compatibility support.